test/widgets/button_test.dart

```diff
@@ -0,0 +1,69 @@
+import 'dart:math';
+
+import 'package:checks/checks.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:flutter/material.dart';
+import 'package:legacy_checks/legacy_checks.dart';
+import 'package:zulip/widgets/button.dart';
+
+import '../flutter_checks.dart';
+import '../model/binding.dart';
+import 'test_app.dart';
+import 'text_test.dart';
+
+
+void main() {
+  TestZulipBinding.ensureInitialized();
+
+  group('ZulipWebUiKitButton', () {
+    final textScaleFactorVariants = ValueVariant(Set.of(kTextScaleFactors));
+    testWidgets('button and touch-target heights', (tester) async {
+      addTearDown(testBinding.reset);
+      tester.platformDispatcher.textScaleFactorTestValue = textScaleFactorVariants.currentValue!;
+      addTearDown(tester.platformDispatcher.clearTextScaleFactorTestValue);
+
+      final buttonFinder = find.byType(ZulipWebUiKitButton);
+
+      int numTapsHandled = 0;
+      await tester.pumpWidget(TestZulipApp(
+        child: UnconstrainedBox(
+          child: ZulipWebUiKitButton(
+            label: 'Cancel',
+            onPressed: () => numTapsHandled++))));
+      await tester.pump();
+
+      final element = tester.element(buttonFinder);
+      final renderObject = element.renderObject as RenderBox;
+      final width = renderObject.size.width;
+      final buttonTopLeft = tester.getTopLeft(buttonFinder);
+      final buttonCenter = tester.getCenter(buttonFinder);
+      check(element).size.isNotNull().height.equals(44); // includes outer padding
+
+      // Outer padding responds to taps, not just the painted part.
+      int numTaps = 0;
+      for (double y = 0; y < 44; y++) {
+        await tester.tapAt(Offset(buttonCenter.dx, y + buttonTopLeft.dy));
+        numTaps++;
+      }
+      check(numTapsHandled).equals(numTaps);
+
+      final textScaler = TextScaler.linear(textScaleFactorVariants.currentValue!)
```

Comment:
I think this test would break up pretty cleanly into two tests at this spot: one checks the response to taps, and one checks the painting. That would probably make them both a little easier to read.